Physical_Engine/Query/HasMergeablePropertiesWith.cs

```diff
@@ -0,0 +1,119 @@
+﻿/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2020, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+using BH.oM.Reflection.Attributes;
+using System.ComponentModel;
+
+using BH.oM.Physical.Elements;
+using BH.oM.Diffing;
+using BH.Engine.Geometry;
+
+namespace BH.Engine.Physical
+{
+    public static partial class Query
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Evaluates if the two IFramingElements' non-geometrical data is equal to the point that they could be merged into one object.")]
+        [Input("element", "An IFramingElement to compare the properties of with an other IFramingElement.")]
+        [Input("other", "The IFramingElement to compare with the other IFramingElement.")]
+        [Output("equal", "True if the IFramingElements' non-geometrical property is equal to the point that they could be merged into one object.")]
+        public static bool HasMergeablePropertiesWith(this IFramingElement element, IFramingElement other)
```

Comment:
Now that we check tangents and normals for this, the framing elements need to have a curve defined, to not crash out with null reference exceptions. Need to check for that before those methods are being called. (if any curve is null, return false, as adding curve might make it false when it comes to normals and tangents)